File: 1356-sort-integers-by-the-number-of-1-bits/1356-sort-integers-by-the-number-of-1-bits.cpp
class Solution {
public:
    vector<int> sortByBits(vector<int>& arr) {
        sort(begin(arr), end(arr), [](int a, int b)->int {
            int aCount = __builtin_popcount(a), bCount = __builtin_popcount(b);
            if (aCount == bCount)
                return (a < b);
            return (aCount < bCount);
        });
        return (arr);
    }
};

File: 1356-sort-integers-by-the-number-of-1-bits/1356-sort-integers-by-the-number-of-1-bits.py
import sortedcontainers;
class Solution(object):
    def sortByBits(self, arr):
        result = [];
        for index, num in enumerate(arr):
            binaryRep = bin(num);
            setBitCount = binaryRep.count('1');
            heapq.heappush(result, (setBitCount, num));
        i = 0;
        for _ in range(len(result)):
            item = heapq.heappop(result);
            arr[i] = item[1];
            i+=1;
        return (arr);
            